frontend/src/metabase/admin/permissions/selectors/data-permissions/permission-editor.ts

```diff
@@ -280,6 +281,17 @@ export const getDatabasesPermissionEditor = createSelector(
     const breadcrumbs = getDatabasesEditorBreadcrumbs(params, metadata, group);
     const title = t`Permissions for the `;
 
+    const deprecatedPermsInGraph = new Set(
+      _.compact([
+        hasPermissionValueInEntityGraphs(
+          permissions,
+          entities.map((entity: any) => ({ groupId, ...entity.entityId })),
+          DataPermission.VIEW_DATA,
+          DataPermissionValue.LEGACY_NO_SELF_SERVICE,
+        ) && DataPermissionValue.LEGACY_NO_SELF_SERVICE,
+      ]),
+    );
```

Comment:
nit: if I understand right, this Set either has one element or none - is that right? If so, I wonder if there's a simpler approach